src/ws_deque.mli

```diff
@@ -1,55 +1,156 @@
-    is efficient work distribution. Owner uses [push] and [pop] method to operate at
-    one end of the deque, while other (free) cores can efficiently steal work on the
-    other side.
+      The main strength of a deque in a typical work-stealing setup with a 
+      per-core structure, is efficient work distribution. The owner uses [push] 
+      and [pop] methods to operate at one end of the deque, while other (free) 
+      cores can efficiently steal work from the other side.
 
-    This approach is great for throughput. Stealers and owner working on different sides
-    reduces contention in work distribution. Further, local LIFO order runs related tasks
-    one after one improves locality.
+      This approach is great for throughput. Stealers and the owner working on
+      different sides, reduce contention in work distribution. Further, the 
+      local LIFO order, running related tasks one after another, improves locality.
 
-    On the other hand, the local LIFO order does not offer any fairness guarantees.
-    Thus, it is not the best choice when tail latency matters.
+      On the other hand, the local LIFO order does not offer any fairness 
+      guarantees. Thus, it is not the best choice when tail latency matters.
 *)
 
+(** {1 API} *)
+
 type 'a t
 (** Type of work-stealing queue *)
 
 val create : unit -> 'a t
 (** [create ()] returns a new empty work-stealing queue. *)
 
+val of_list : 'a list -> 'a t
+(** [of_list list] creates a new work-stealing queue from [list].
+
+  🐌 This is a linear-time operation.
+
+ {[
+        # open Saturn.Work_stealing_deque
+        # let t : int t = of_list [1;2;3;4]
+        val t : int t = <abstr>
+        # pop_opt t
+        - : int option = Some 4
+        # pop_opt t 
+        - : int option = Some 3
+ ]}
+*)
+
 exception Empty
 
-(** {1 Queue owner functions} *)
+(** {2 Queue owner functions} *)
 
 val push : 'a t -> 'a -> unit
-(** [push t v] adds [v] to the front of the queue [q].
-      It should only be invoked by the domain which owns the queue [q]. *)
+(** [push queue element] adds [element] at the end of the [queue].
+      It should only be invoked by the domain that owns the [queue]. *)
 
 val pop_exn : 'a t -> 'a
-(** [pop_exn q] removes and returns the first element in queue
-      [q].It should only be invoked by the domain which owns the queue
-      [q].
+(** [pop_exn queue] removes and returns the last element of the [queue]. It 
+      should only be invoked by the domain that owns the [queue].
 
-      @raise Exit if the queue is empty.
-      *)
+       @raises Empty if the [queue] is empty. *)
 
 val pop_opt : 'a t -> 'a option
-(** [pop_opt q] removes and returns the first element in queue [q], or
-    returns [None] if the queue is empty.  *)
+(** [pop_opt queue] removes and returns [Some] of the last element of the 
+      [queue], or returns [None] if the [queue] is empty.  *)
```

Comment:
Not sure what's the best documentation practice here, but just in case: Considering that many users access documentation through hovering symbols in their text editor, perhaps it's better to repeat the warning about which domain can invoke the function? (but this does lead to repetitive odoc pages so I don't know if that's better)

Reply:
Absolutely ! Thanks !